source/MaterialXView/Handlers/GeometryHandler.h

```diff
@@ -0,0 +1,82 @@
+#ifndef MATERIALX_GEOMETRYHANDLER_H
+#define MATERIALX_GEOMETRYHANDLER_H
+
+#include <string>
+#include <memory>
+
+namespace MaterialX
+{
+/// Shared pointer to an GeometryHandler
+using GeometryHandlerPtr = std::shared_ptr<class GeometryHandler>;
+
+/// @class @GeometryHandler
+/// Abstract class representing a geometry handler for providing data for shader binding.
+///
+class GeometryHandler
+{
+  public:
+    /// Default constructor
+    GeometryHandler();
+    
+    /// Default destructor
+    virtual ~GeometryHandler();
+
+    /// Set the identifier which will indicate what geometry to return
+    /// e.g. SCREEN_ALIGNED_QUAD is one identifier
+    virtual void setIdentifier(const std::string identifier);
+    
+    /// Get geometry identifier
+    const std::string& getIdentifier() const
+    {
+        return _identifier;
+    }
+
+    /// Create indexing data for geometry. The indexing is assumed to 
+    /// be for a set of triangles. That is every 3 values index triangle data.
+    /// @param bufferSize The size of the returned buffer in bytes.
+    /// @param indexCount Number of indices in the indexing
+    virtual unsigned int* getIndexing(size_t &bufferSize, unsigned int& indexCount) = 0;
+
+    /// Create position data. Each element is assumed to be 3 float in size.
+    /// @param bufferSize The size of the returned buffer in bytes.
+    /// @param screenWidth Width of screen. Required to initialzie screen positions. Defaults to 0.
+    /// @param screenHeight Height of screen. Required to initialzie screen positions. Defaults to 0.
+    virtual float* getPositions(size_t &bufferSize, unsigned screenWidth=0, unsigned int screenHeight=0) = 0;
```

Comment:
I wonder if screen size belongs in this interface. I know you need it to create the screen aligned quad, but isn't that a special case of geometry? Should all geometry created by a geometry handler be done in "screen space"?

Reply:
Yeah. I don't like this.I was thinking of add in a  getScreenQuadPositions(). How does that sound?

Reply:
Another option would be to add in a class holding global properties (where screen resolution could be one). Either a singleton or something you pass along. You don't have to do that now though, we can add a separate task for that.